babyagi.py

```diff
@@ -12,32 +12,24 @@
 #Set Variables
 load_dotenv()
 
-# Set API Keys
-OPENAI_API_KEY = os.getenv("OPENAI_API_KEY", "")
-assert OPENAI_API_KEY, "OPENAI_API_KEY environment variable is missing from .env from .env"
+# get environment variables
+def get_env_var(var_name, error_message):
+    var_value = os.getenv(var_name, "")
+    assert var_value, f"{var_name} environment variable is missing from .env: {error_message}"
```

Comment:
i don't think you should use an empty string as the default.  what you're checking here is that there is a variable at all, and if there isn't, the result here would be `None` (you could set an explicit `default=None`, but that's already the default behavior).  then use a more explicit check that the variable `is not None` in the following `assert`.

```suggestion
    var_value = os.getenv(var_name)
    assert var_value is not None, f"{var_name} environment variable is missing from .env: {error_message}"
```

Reply:
i also don't know if you really need the `error_message` arg, since you're just passing in the same thing each time.  i'd scrap that piece.
```suggestion
    var_value = os.getenv(var_name)
    assert var_value is not None, f"{var_name} environment variable is missing from .env"
```